File: sanidhyamangal/binary_search.py
# function to perform binary search
def binary_search(array: list, num:int) -> tuple:
    """
    function to perform binary search on the given array
    :param array: sorted array on which searching is needed to perform
    :param num: number which is required to found
    :return: tuple of position and number
    """
    # apply merge sort to the function 
    # find out the values of high and low
    low,high = 0, len(array)
    # iterate until low is less than high
    while low <= high:
        # find the mid of the array
        mid = (low + high) // 2
        # check if mid of array is equal to number
        if num == array[mid]:
             # return mid and num
            return mid, num
        # if num is less than mid of array update high as mid-1 of array
        elif num < array[mid]:
            high = mid-1
        # else update low as mid+1 of array
        else:
            low = mid + 1
    # return none for the position of the number
    return None, num

File: sanidhyamangal/binary_search.cpp
#include<iostream>
#include<vector>

using namespace std;

int binary_search(vector<int> array, int search_term){
   int low =0, high = array.size();

   while (low <= high)
   {
       int mid = (low + high) / 2;
       if (search_term == array[mid]){
           return mid;
       }
       else if (search_term > array[mid]){
           low = mid +1;
       }
       else{
           high = mid -1;
       }
   }

   return -1;
}

int main(){
    vector<int> temp = vector<int>{2,3,5,6,9};
    cout << binary_search(temp, 6);
    return 0;
}